apps/modernization-ui/src/apps/deduplication/data-elements/form/DataElementRow/DataElementRow.tsx

```diff
@@ -0,0 +1,150 @@
+                    oddsRatio: configuration?.[field]?.oddsRatio,
+                    logOdds: 0, // Will be recalculated when oddsRatio changes
+                    threshold: configuration?.[field]?.threshold
+                };
+                form.setValue(field, defaultValue);
+            }
+        }
+    }, [watch[field]?.active]);
+
+    useEffect(() => {
+        const oddsRatio = Number(watch[field]?.oddsRatio);
+
+        const isValidOddsRatio = !isNaN(oddsRatio) && oddsRatio > 0 && oddsRatio < 1;
+
+        if (!isValidOddsRatio) {
+            form.setValue(`${field}.logOdds`, undefined);
+            return;
+        }
+
+        form.setValue(`${field}.logOdds`, Math.log(oddsRatio));
+    }, [watch[field]?.oddsRatio]);
+
+    return (
+        <tr>
+            <td className={styles.checkbox}>
+                <Controller
+                    control={form.control}
+                    name={`${field}.active`}
+                    render={({ field: { value, onChange } }) => (
+                        <Checkbox
+                            name={`${field}.active`}
+                            label=""
+                            id={`${field}-checkbox`}
+                            selected={value}
+                            onChange={onChange}
+                            aria-label={fieldName}
+                            role="checkbox"
+                            aria-checked={value}
+                            aria-labelledby={`${field}-checkbox-label`}
+                        />
+                    )}
+                />
+            </td>
+            <td className={styles.field}>{fieldName}</td>
+            <td>
+                <Controller
+                    control={form.control}
+                    name={`${field}.oddsRatio`}
+                    rules={{
+                        required: {
+                            value: watch[field]?.active ?? false,
+                            message: 'Missing odds ratio'
+                        },
+                        validate: (value) => {
+                            if (String(value) === '') return true; // Allow empty input to clear error
+                            const numericValue = Number(value);
+                            if (isNaN(numericValue) || numericValue <= 0 || numericValue >= 1) {
+                                return 'Requires a value between 0>1';
+                            }
+                            return true;
+                        }
+                    }}
+                    render={({ field: { value, onChange, onBlur, name }, fieldState: { error } }) => (
+                        <TableNumericInput
+                            name={name}
+                            value={value ?? ''}
+                            onChange={(e) => {
+                                onChange(e); // Normal onChange event
+                                form.trigger(`${field}.oddsRatio`); // Manually trigger validation
```

Comment:
Suggestion: I think you can just trigger the `onBlur()` method here since form is set up to validate on blur events.